api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -252,5 +256,40 @@ else if (!isObsGroup) {
 			}
 		}
 	}
+
+	/**
+	 * This method validates Obs' concept values.
+	 * 
+	 * <ol>
+	 *     <li>Validates if high absolute and low absolute are within the valid range</li>
+	 *     <li>Validates if patient's age is within the valid range</li>
+	 * <ol/>
+	 * in the {@link ConceptReferenceRange}.
+	 *
+	 * @param obs Observation to validate
+	 * @param errors Errors to record validation issues
+	 */
+	private void validateConceptReferenceRange(Obs obs, Errors errors) {
+		Concept concept = obs.getConcept();
+
+		if (concept != null && concept.getDatatype() != null && concept.getDatatype().isNumeric()) {
+			Optional<ConceptReferenceRange> crr = Context.getConceptService()
+				.getConceptReferenceRangeByConceptId(concept.getConceptId());
```

Comment:
Since a concept can have more than one reference range with different criteria, How do you filter out the other reference ranges?

Reply:
I think this follows your previous question above. I am open to your suggestion.